### ISSUE TITLE
Query: Can you scaffold directly against Sql Db Project

### ISSUE THREAD

**Issue by rposener** [dotnet/efcore#10837]
We have a large database project that we use to manage the SQL Database.  Currently we have to do a deploy with special options like make sure it removes all extra tables, etc. before we can scaffold.  Could a scaffold process not be created to scaffold directly from the dacpac project instead of having to jump through the hoops of deploying to SQL then scaffolding back to our C# project?

**Comment by ErikEJ** [comment]
Yes, potentially you could implement you own IDatabaseModelFactory and use DacFX to generate it. 

https://the.agilesql.club/Blogs/Ed-Elliott/DacFx-Create-tSQLt-Tests-From-A-Dacpac

https://github.com/aspnet/EntityFrameworkCore/blob/dev/src/EFCore.SqlServer/Scaffolding/Internal/SqlServerDatabaseModelFactory.cs

Would make a nice addition to "EF Core Power Tools"! (of course also a lib/command line tool)

**Comment by bricelam** [comment]
Duplicate of #4321

**Comment by bricelam** [comment]
@ErikEJ is correct.

The end-to-end experience could look something like this:
```
dotnet install package EFCore.SqlServer.Dacpac
dotnet ef dbcontext scaffold ..\TheDatabase\bin\Debug\TheDatabase.dacpac EFCore.SqlServer.Dacpac
```

**Comment by bricelam** [comment]
Where EFCore.SqlServer.Dacpac would basically use all the services directly from Microsoft.EntityFrameworkCore.SqlServer but replace the `IDatabaseModelFactory`.

**Comment by ErikEJ** [comment]
Currently Windows only, though

**Comment by ErikEJ** [comment]
@rposener This feature is now available in the latest daily build of "EF Core Power Tools" https://github.com/ErikEJ/SqlCeToolbox/wiki/EF-Core-Power-Tools

**Comment by ErikEJ** [comment]
Is this syntax also possible with PMC, and if yes, what does it look like:

    dotnet ef dbcontext scaffold ..\TheDatabase\bin\Debug\TheDatabase.dacpac EFCore.SqlServer.Dacpac

**Comment by bricelam** [comment]
Should just be...
``` ps1
Scaffold-DbContext ..\TheDatabase\bin\Debug\TheDatabase.dacpac EFCore.SqlServer.Dacpac
```

**Comment by bricelam** [comment]
The relative path is weird though. It's probably relative to the target project, not the current directory.

**Comment by ErikEJ** [comment]
I Will document a full path!

**Comment by bricelam** [comment]
It's a similar issue to #10036, but probably requires a different solution since yours is inside a `connectionString`.
